src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/EmbeddingProvider.java

```diff
@@ -9,6 +10,8 @@
  * of chosen model.
  */
 public interface EmbeddingProvider {
```

Comment:
This may be better idea for follow-up PR, not here, but I think that instead of interface, we should make this an abstract class that takes `String providerId`, defines abstract method `String getErrorMessage(response)` and can then implement `mapException()` in this class.
This would reduce copied code quite bit.

But need not be done within this PR, just mentioning it (I could do PR once this is merged).

Reply:
Created a new issue about this: https://github.com/stargate/data-api/issues/1173